### ISSUE TITLE
Exception for http site when 'Block all unencrypted requests' enabled

### ISSUE THREAD

**Issue by PekaFat** [EFForg/https-everywhere#8854]
Hi, how i can add exception for site that not support https but 'Block all unencrypted requests' is enabled?

**Comment by jeremyn** [comment]
@PekaFat Thanks for your report. There's no way to do this in HTTPS Everywhere and I don't think we want to add it as an option either, sorry. I think it would be too confusing. One thing you might do outside of HTTPS Everywhere is create a new profile with your browser (see [here](https://support.mozilla.org/t5/Install-and-Update/Use-the-Profile-Manager-to-create-and-remove-Firefox-profiles/ta-p/2914) for Firefox) and only use the separate profile for the site that doesn't work.

If you use "Block all unencrypted requests" because you want HTTPS coverage on certain sites that currently aren't covered, please feel free to request that we add those sites (or better, submit a pull request to add them yourself).

@Hainish Do you agree with me here?

**Comment by Hainish** [comment]
Yeah we don't want to start adding exceptions for this feature.  The whole point is that it blocks sites which don't offer encryption.

**Comment by crisdavid7341** [comment]
Is there a way you can make an exception via the HTTPS Everywhere extension; to not run on a site if you don't have the "block all unencrypted requests" feature on?

**Comment by regumuricu** [comment]
I was very pleased to see that just recently the changelog of HTTPS Everywhere showed this:

```
Firefox 5.2.9 / Chrome 2016.12.19
  * In HTTP Nowhere mode, attempt HTTPS before block
```

It's a feature I've always wanted and I'm really glad you finally added it.

However, I'm with PekaFat on this suggestion to add a whitelist. And allow me to explain as I am a user who is in HTTP Nowhere mode 100% of the time. The reason I like HTTP Nowhere is because I can browse freely through the web with the confidence in knowing that all requests will be HTTPS. Whereas in HTTPS Everywhere mode this confidence is lost because I don't know what will happen with the next page I visit. Plus even if I had rules for a site in HTTPS Everywhere, these sites can change without warning adding new domains or whatever so one day I could leak and I always have to make sure my rules are up to date and even if they are up to date it could be a new change that hasn't yet been added to HTTPS Everywhere. It's just an unfortunate thing to deal with in HTTPS Everywhere.

So the solution for me is HTTP Nowhere. This solves my worries and restores my confidence. However, the side effect is now I have a new problem. Everyday browsing is hard to do because there's always some site that doesn't have HTTPS and I have to keep disabling/re-enabling HTTP Nowhere many times per browsing session. It's really annoying especially with some of my daily sites that I visit which do have HTTPS but have random pictures hosted on image hosts that don't support HTTPS, these pictures can't be seen because they get blocked by HTTP Nowhere. For example, imagebam.com is a popular one that I must whitelist because I see it every day and they have no SSL!

In HTTP Nowhere mode there must be a whitelist in order to make everyday browsing possible.

@Hainish The whitelist is an optional feature for those who find it useful. By default the whitelist is empty, so users who don't need a whitelist will not see any difference in the behavior they expect because with an empty whitelist HTTP Nowhere will continue to operate as it currently does. There is no harm in adding it as it only gives more flexibility to people like me and PekaFat.

I currently use a different addon called HTTP Nowhere:

https://github.com/cwilper/http-nowhere
https://addons.mozilla.org/firefox/addon/http-nowhere/

It's the best, but it hasn't been updated in years and occasionally it's a little buggy. It has the whitelist and it does work nicely but being so outdated it has it's moments when it acts strangely. But for the most part I've been using it for the past 3 years and really love it.

But when I saw the latest changelog of HTTPS Everywhere saying it now "attempts HTTPS before block" I thought to myself, FINALLY! Now we're getting somewhere. But without the whitelist it's just not enough. Please add the whitelist, it does no harm and only gives the users more ways to use the great HTTPS Everywhere addon.

I don't like adding rules and updating rules and worrying about which sites are leaking and browsing obscure sites that will never have rules and all those things... I just want to browse peacefully in HTTP Nowhere mode, knowing that nothing slips by unencrypted unless I whitelist it.

Thank you.

**Comment by MarcSN311** [comment]
I second the Whitelist request because otherwise 'Block all unencrypted requests' breaks other Security improving Add-Ons, namely PassIFox (the Firefox extension to use KeePass as Password storage backend).

**Comment by Hainish** [comment]
I'll keep this issue open, it might be a good task for a volunteer.  Currently we can't dedicate cycles to it but we do accept contributions :)

**Comment by jeremyn** [comment]
@MarcSN311 The PassIFox problem is better served by issue #8242, which is asking for a specific, permanent exception to rewriting localhost to HTTPS when "Block all unencrypted requests" is enabled. Other than localhost connections I can't think of any use-case for a security add-on intended for normal browsing to need an exception to the "Block all" rule. If you or anyone else has such a use-case, please describe it here.

**Comment by zneaa** [comment]
Duplicate https://github.com/EFForg/https-everywhere/issues/2299

**Comment by strugee** [comment]
@crisdavid7341 if you click on the HTTPS Everywhere icon it will show you rulesets enabled for the current page. You can click on rulesets to disable or enable them.

**Comment by Hainish** [comment]
Thanks @zneaa, closing this in favor of the older #2299
